### PR TITLE
Generalized security scan

### PR DESCRIPTION
**Pull request by mlholland**
Takes code from the following files:
 - https://github.com/solo-io/gloo/blob/master/docs/cmd/securityscanutils/securityscan_utils.go
 - https://github.com/solo-io/gloo/blob/master/docs/cmd/generate_docs.go

... And merges them into a single CLI that has a bunch of extra inputs to make it completely repo-agnostic.
BOT NOTES: 
resolves https://github.com/solo-io/go-utils/issues/467

### CODE REVIEW

File: securityscanutils/cli/exampleVersionImageConstraints.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# comments are skipped, as are empty lines
+
+constraint-a , image1, image2, image3
```
Comment (saiskee) [inline securityscanutils/cli/exampleVersionImageConstraints.txt:3]: the spaces will be included in the image names, which may cause errors here. Do you strip the spaces before processing the csv?

Comment (saiskee) [reply securityscanutils/cli/exampleVersionImageConstraints.txt:3]: Also can you include an actual example of images and a constraint, like
```csv
>=v1.6.0, gloo-ee, rate-limit-ee, extauth-ee
```

Comment (mlholland) [reply securityscanutils/cli/exampleVersionImageConstraints.txt:3]: Whitespace is trimmed here: https://github.com/solo-io/go-utils/pull/468/files#diff-f92cc044bef1930064629dba7b446d6e84fabfb02082830689bcc858f6e797dcR398

Comment (mlholland) [reply securityscanutils/cli/exampleVersionImageConstraints.txt:3]: Added a commented real example constraint line.

**Review by saiskee — APPROVED** [review]
lgtm! Really excited to have this logic extracted out

**Pull request merged**